--- origsrc/telepathy-mission-control-5.10.1/mission-control-plugins/Makefile.am	2011-10-04 09:28:30.000000000 -0500
+++ src/telepathy-mission-control-5.10.1/mission-control-plugins/Makefile.am	2011-11-20 00:37:42.822399400 -0600
@@ -33,6 +33,7 @@ plugins_include_HEADERS = \
 
 libmission_control_plugins_la_LDFLAGS = \
 	-export-symbols-regex "^(mcp_)" \
+	-no-undefined \
 	-version-info "@MCP_LT_CURRENT@":"@MCP_LT_REVISION@":"@MCP_LT_AGE@"
 
 libmission_control_plugins_la_LIBADD = \
