--- origsrc/telepathy-mission-control-5.2.6/libmcclient/Makefile.am	2009-08-19 09:28:11.000000000 -0500
+++ src/telepathy-mission-control-5.2.6/libmcclient/Makefile.am	2009-10-29 12:40:17.840595000 -0500
@@ -80,7 +80,7 @@ libmcclient_la_LIBADD = \
 	$(GLIB_LIBS)
 
 libmcclient_la_LDFLAGS = \
-	-export-symbols-regex "^(mc_)|(mission_control_)"
+	-export-symbols-regex "^(mc_)|(mission_control_)" -no-undefined
 
 if ENABLE_MC_CLIENT_SO6
 # the redundant quoting here is to prevent the libtool command line from
--- origsrc/telepathy-mission-control-5.2.6/src/Makefile.am	2009-10-08 06:47:45.000000000 -0500
+++ src/telepathy-mission-control-5.2.6/src/Makefile.am	2009-10-29 12:40:41.890628200 -0500
@@ -81,7 +81,7 @@ libmissioncontrol_server_la_SOURCES =
 libmissioncontrol_server_la_LIBADD = libmcd-convenience.la
 
 libmissioncontrol_server_la_LDFLAGS = \
-	-export-symbols-regex '^((mc_)|(mcd_)|(mission_control_))'
+	-export-symbols-regex '^((mc_)|(mcd_)|(mission_control_))' -no-undefined
 
 if ENABLE_MC_SERVER_SO6
 # the redundant quoting here is to prevent the libtool command line from
