--- origsrc/telepathy-mission-control-4.65/libmcclient/Makefile.am	2008-04-08 06:25:54.000000000 -0500
+++ src/telepathy-mission-control-4.65/libmcclient/Makefile.am	2008-06-24 23:26:39.906250000 -0500
@@ -52,6 +52,7 @@
 
 libmcclient_la_LDFLAGS = $(common_ldflags) \
 				      -export-symbols-regex "^(mc_)|(mission_control_)" \
+				      -no-undefined \
 				      -version-info 5:0:0
 
 %-marshal.h: %-marshal.list Makefile.am
--- origsrc/telepathy-mission-control-4.65/libmissioncontrol/Makefile.am	2008-04-08 06:26:35.000000000 -0500
+++ src/telepathy-mission-control-4.65/libmissioncontrol/Makefile.am	2008-06-24 23:55:01.171875000 -0500
@@ -80,6 +80,7 @@
 
 libmissioncontrol_client_la_LDFLAGS = $(common_ldflags) \
 				      -export-symbols-regex "^(mc_)|(mission_control_)" \
+				      -no-undefined \
 				      -version-info 4:2:4
 
 mission_control_include = \
@@ -90,10 +91,10 @@
 	dbus-binding-tool --prefix=mc-client-lib --mode=glib-client $< > $@
 
 %-marshal.h: %-marshal.list Makefile.am
-	glib-genmarshal --header --prefix=$(subst -,_,$*)_marshal $< > $*-marshal.h
+	glib-genmarshal --header --prefix=$(subst -,_,$(*F))_marshal $< > $(*F)-marshal.h
 
 %-marshal.c: %-marshal.list Makefile.am
-	glib-genmarshal --body --prefix=$(subst -,_,$*)_marshal $< > $*-marshal.c
+	glib-genmarshal --body --prefix=$(subst -,_,$(*F))_marshal $< > $(*F)-marshal.c
 
 mc-enum-types.h: stamp-mc-enum-types.h
 	@true
--- origsrc/telepathy-mission-control-4.65/src/Makefile.am	2008-04-08 06:25:55.000000000 -0500
+++ src/telepathy-mission-control-4.65/src/Makefile.am	2008-06-24 23:27:09.968750000 -0500
@@ -68,6 +68,7 @@
 
 # we want to export symbols so that the plugins can see them
 libmissioncontrol_server_la_LDFLAGS = -export-dynamic \
+				      -no-undefined \
 				      -version-info 4:0:3
 
 libmissioncontrol_server_la_SOURCES = \
